tests/test_caching.py

```diff
@@ -199,6 +199,28 @@ def test_cache_window(self):
 
         self.assertEqual(len(trace1.sched_wakeup.data_frame), 2)
 
+    def test_cache_window_shift(self):
+        """Test that cache is invalidated when window is shifted"""
+        GenericFTrace.disable_cache = False
+
+        trace1 = trappy.FTrace(
+            events=['sched_wakeup'],
+            window=(0, 1))
+
+        if len(trace1.sched_wakeup.data_frame) != 1:
+            raise RuntimeError('Test bug: bad sched_wakeup event count')
+
+        # Parse again with a shifted window
+        trace2 = trappy.FTrace(
+            events=['sched_wakeup'],
+            window=(1, 2))
+
+        # Cache should have been invalidated to fetch data after first window
+        self.assertGreater(
+            trace2.sched_wakeup.data_frame.index[-1],
+            trace1.sched_wakeup.data_frame.index[-1]
+        )
```

Comment:
Ah, I didn't spot this, it should probably go in the same commit as the functionality it tests. This coverage definitely helps, but as I said I'd rather be really conservative here and only go for the window checking if we have tested multiple corners of the input space.